cli/src/cli.rs

```diff
@@ -0,0 +1,80 @@
+use crate::config::{Config, FromFile, LoggerConfig};
+use clap::{crate_authors, crate_version, Parser};
+use error_stack::ResultExt;
+pub use error_stack::{Context, Report};
+use std::{fmt::Display, path::PathBuf};
+
+lazy_static::lazy_static! {
+    static ref VERSION: String = {
+        format!(
+            concat!("BOB-GUI VERSION: {}\n",
+            "BUILT AT: {}\n",
+            "COMMIT HASH: {}\n",
+            "GIT BRANCH/TAG: {}\n"),
+            crate_version!(),
+            build_time::build_time_utc!(),
+            option_env!("GIT_HASH").unwrap_or("-"),
+            option_env!("GUI_BUILD_BRANCH_TAG").unwrap_or("-"),
+        )
+    };
+}
+
+/// Bob configuration
+#[derive(Debug, Parser, Clone)]
+#[command(author = crate_authors!())]
+#[command(version = VERSION.trim(), about, long_about)]
+#[group(id = "configs", required = true, multiple = false)]
+pub struct Args {
+    /// If set, passes default configuration to the server
+    #[clap(short, long)]
+    default: bool,
+
+    /// Server configuration file
+    #[arg(short, long, value_name = "FILE")]
+    config_file: Option<PathBuf>,
+}
+
+impl TryFrom<Args> for Config {
+    type Error = Report<Error>;
+
+    fn try_from(value: Args) -> Result<Self, Self::Error> {
+        if value.default {
+            Ok(Self::default())
+        } else if let Some(config) = value.config_file {
+            Self::from_file(config).change_context(Error::Config)
+        } else {
+            unreachable!()
+        }
+    }
+}
+
+impl TryFrom<Args> for LoggerConfig {
```

Comment:
This implementation is redundant. The only way to get the logger config is to load the application config (`Config`) from file and get logger config from it. We are not currently considering the option of placing the logger config in a separate file